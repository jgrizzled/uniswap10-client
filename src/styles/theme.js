export default {
  fontSize: {
    small: '0.85rem',
    normal: '1rem',
    large: '1.25rem',
    xLarge: '1.5rem',
    xxLarge: '2rem'
  },
  font: {
    main: "'Roboto', sans-serif;"
  },
  fontWeight: {
    normal: '400',
    bold: '700'
  },
  space: {
    small: '0.5rem',
    medium: '1rem',
    large: '2rem',
    extraLarge: '4rem'
  },
  color: {
    primary: '#2373CC',
    primaryLight: '#4989D3',
    background: '#f2f2f2',
    surface: '#ffffff',
    error: '#B00020',
    onPrimary: '#FFFFFF',
    onSecondary: '#000000',
    onBackground: '#000000',
    onSurface: '#000000',
    onError: '#FFFFFF',
    link: '#1890ff'
  },
  breakpoint: {
    desktop: '40em'
  }
};
