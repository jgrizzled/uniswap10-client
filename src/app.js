// Main app component

import React, { useState, useEffect } from 'react';
import { Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import GlobalStyle from 'styles/GlobalStyle';
import theme from 'styles/theme';
import Header from 'components/header';
import IndexOverview from 'components/pages/index-overview';
import Loading from 'components/loading';
import BurgerMenu from 'components/burger-menu';
import FAQ from 'components/pages/faq';
import Contact from 'components/pages/contact';
import Settings from 'components/pages/settings';
import { fetchIndexData } from 'api';

export default function App() {
  const [indexData, setIndexData] = useState(null);
  const [indexSettings, setIndexSettings] = useState({
    currency: 'usd',
    liquidityWeight: '0.5',
    rebalancePeriod: '30',
  });

  // fetch index data on new settings
  useEffect(() => {
    const fetchData = async () => {
      try {
        const { indexByDate, holdingsByDate, tokens } = await fetchIndexData(
          indexSettings
        );
        setIndexData({ indexByDate, holdingsByDate, tokens });
      } catch (e) {
        console.error(e);
      }
    };
    fetchData();
    // eslint-disable-next-line
  }, [JSON.stringify(indexSettings)]);

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <Header />
      <BurgerMenu />
      <Switch>
        <Route path='/settings'>
          <Settings settings={indexSettings} setSettings={setIndexSettings} />
        </Route>
        <Route path='/faq'>
          <FAQ />
        </Route>
        <Route path='/contact'>
          <Contact />
        </Route>
        <Route>
          {indexData ? <IndexOverview data={indexData} /> : <Loading />}
        </Route>
      </Switch>
    </ThemeProvider>
  );
}
