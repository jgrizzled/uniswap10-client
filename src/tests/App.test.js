import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from '../app';
import theme from 'styles/theme';

const app = <App themes={theme} />;

// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(app, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', () => {
  const tree = renderer.create(app).toJSON();
  expect(tree).toMatchSnapshot();
});
