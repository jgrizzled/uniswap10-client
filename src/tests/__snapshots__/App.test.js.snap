// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
Array [
  <header
    className="sc-AxjAm OAJFQ"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        Uniswap 10
      </h1>
    </a>
  </header>,
  <div
    className="sc-fzoyAV gRtasW"
  >
    <button
      className="sc-fzqBZW befhGh"
      onClick={[Function]}
      open={false}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <nav
      className="sc-fzqNJr cneoWL"
      open={false}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        <span
          aria-label="settings"
          role="img"
        >
          ðŸ› 
        </span>
        Settings
      </a>
      <a
        href="/faq"
        onClick={[Function]}
      >
        <span
          aria-label="faq"
          role="img"
        >
          ðŸ¤”
        </span>
        FAQ
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        <span
          aria-label="contact"
          role="img"
        >
          ðŸ“©
        </span>
        Contact
      </a>
    </nav>
  </div>,
  <h2
    className="sc-fzokOt emxQnu"
  >
    Loading
  </h2>,
]
`;
